### PR TITLE
feat(audio): custom surround-params

### PR DESCRIPTION
**Pull request by mariotaku**
## Description

This PR is made for improving webOS client support. On webOS 5.0 and above, if surround channel configuration is done properly, it can re-encode OPUS stream on-the-fly & send it to surround sound system through eARC/Optical.

Currently, webOS only handles 6-channels, 4 streams & 2 coupled streams configuration, and the channel must be ordered as `[0, 1, 4, 5, 2, 3]` which Sunshine doesn't have. It won't accept stream header either, which makes this PR necessary.

With this PR, if `surroundParams` is specified (formatted like `surround-params=` in RTSP handshake, e.g. `642014523`) in HTTP launch request, it will be applied, and during RTSP handshake, our custom surround-params `a=fmtp:97 surround-params=642014523` will be advertised twice first, followed by the original ones.

### Screenshot

Not applicable 

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

File: src/rtsp.cpp [hunk 1]
```diff
@@ -818,6 +818,12 @@ namespace rtsp_stream {
       ss << "a=rtpmap:98 AV1/90000"sv << std::endl;
     }
 
+    if (!session.surround_params.empty()) {
+      // If we have our own surround parameters, advertise them twice first
+      ss << "a=fmtp:97 surround-params="sv << session.surround_params << std::endl;
+      ss << "a=fmtp:97 surround-params="sv << session.surround_params << std::endl;
```
Comment (cgutman) [inline src/rtsp.cpp:824]: What's the purpose of advertising them twice?

Comment (mariotaku) [reply src/rtsp.cpp:824]: The first `surround-params` will be applied to `NormalQualityOpusConfig`, and the second one goes to `HighQualityOpusConfig`. I think advertising them twice first will make sure both configs having the desired mapping.

**Comment by moi952** [comment]
Hi, will ac3 encoding be functional on moonlight-android and moonlight-qt? (provided that development is done I suppose)

**Comment by moi952** [comment]
> @moi952 Hi, this PR won't enable AC3 codec, it only rearranges sound channel mapping. May I ask why you are interested in AC3 codec? As far as I know, all existing clients handles OPUS better.

Thank you for your answer. Too bad, I don't know OPUS but I guess it's PCM?
In fact with my current TV if I connect my Nvidia Shield or my PC to the TV and I put it in 5.1, my amp (Denon AVC-6500x) remains in stereo, I don't know if the TV doesn't allow it to accept uncompressed 5.1.
On the other hand, if I connect my Nvidia Shield directly to my amp I have good sound in 5.1

I tried different sound input/output settings on my TV but it doesn't work.

For example for Windows if I connect my PC to my TV, it does not offer me more than 2 channels in the sound output configuration.

**Comment by ReenigneArcher** [comment]
@mariotaku could you rebase your PR?

**Review by Hazer — APPROVED** [review]

**Comment by ReenigneArcher** [comment]
@mariotaku are you happy with this PR and [tests](https://github.com/LizardByte/Sunshine/pull/2424#issuecomment-2136365298)?

**Comment by mariotaku** [comment]
> @mariotaku are you happy with this PR and [tests](https://github.com/LizardByte/Sunshine/pull/2424#issuecomment-2136365298)?

Yes, I'd like to get it merged :)

**Pull request merged**